Address comments in the main PR.

yoshisatoyanagisawa opened the pull request:
There is ongoing PR review in https://github.com/w3c/ServiceWorker/pull/1701.
I have addressed some of them here.

Until https://github.com/yoshisatoyanagisawa/ServiceWorker/pull/10 is merged, I plan to work here to avoid https://github.com/yoshisatoyanagisawa/ServiceWorker/pull/10 review and update exhausted.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/pull/12.html" title="Last updated on Feb 15, 2024, 5:58 AM UTC (e898871)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/yoshisatoyanagisawa/ServiceWorker/12/5a48d55...e898871.html" title="Last updated on Feb 15, 2024, 5:58 AM UTC (e898871)">Diff</a>

yoshisatoyanagisawa committed:
Reflect some of comments in https://github.com/w3c/ServiceWorker/pull/1701

yoshisatoyanagisawa committed:
Clarified the cache name matching.

yoshisatoyanagisawa committed:
minor fix.

yoshisatoyanagisawa committed:
Run soft update during the router is selected.

yoshisatoyanagisawa committed:
Explicitly unset the flag on terminate.

- also add dfn for agent.

yoshisatoyanagisawa committed:
Use registration's cache map instead of this's.

yoshisatoyanagisawa commented:
@sisidovski @domenic Can I ask you to take a look?

yoshisatoyanagisawa commented:
@mkruisselbrink

mkruisselbrink commented on docs/index.bs:
```diff
@@ -1594,10 +1595,11 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
         1. If |rules| is a {{RouterRule}} dictionary, set |rules| to &#x00AB; |rules| &#x00BB;.
         1.  For each |rule| of |rules|:
-            1. If running [=Verify Router Condition=] algorithm with |rule|["{{RouterRule/condition}}"] and |serviceWorker| returns false, [=throw=] a {{TypeError}}.
+            1. If running [=Verify Router Condition=] algorithm with |rule|["{{RouterRule/condition}}"] and |serviceWorker| returns false, return [=a promise rejected with=] a {{TypeError}}.
```
(as an aside) these changes aren't technically needed since webidl will convert any thrown exceptions into rejected promises anyway, but having them does make it clearer.

yoshisatoyanagisawa replied:
sure.

mkruisselbrink commented on docs/index.bs:
```diff
@@ -3374,8 +3385,7 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. [=list/For each=] |rule| of |serviceWorker|'s [=service worker/list of router rules=]:
-          1. If running [=Match Router Condition=] with |rule|["{{RouterRule/condition}}"], |serviceWorker| and |request| returns false, [=continue=].
-          1. Return |rule|["{{RouterRule/source}}"].
+          1. If running the [=Match Router Condition=] algorithm with |rule|["{{RouterRule/condition}}"], |serviceWorker| and |request|, then return |rule|["{{RouterRule/source}}"].
```
you missed the "returns true" bit

yoshisatoyanagisawa replied:
done.

sisidovski commented on docs/index.bs:
```diff
@@ -3127,6 +3129,7 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
 
           1. [=Abort a running script|Abort the script=] currently running in |serviceWorker|.
           1. Set |serviceWorker|'s [=start status=] to null.
+          1. Unset |serviceWorker|'s [=ServiceWorkerGlobalScope is ready=].
```
Can we also address this comment?
https://github.com/w3c/ServiceWorker/pull/1701/files#r1484868955

yoshisatoyanagisawa replied:
Partially addressed the comment by adding " flag".  However, I believe the main topic of the comment is renaming.
I am considering to call it "ServiceWorkerGlobalScope CSP ready flag"
What do you think?

domenic commented on docs/index.bs:
```diff
@@ -207,7 +208,7 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
     A [=/service worker=] has an associated <dfn>all fetch listeners are empty flag</dfn>. It is initially unset.
 
-    A [=/service worker=] has an associated <dfn>list of router rules</dfn>. It is initially an empty [=list=].
+    A [=/service worker=] has an associated <dfn>list of router rules</dfn> (a [=list=]) whose [=list/item=] is a {{RouterRule}}. It is initially an empty [=list=].
```
```suggestion
    A [=/service worker=] has an associated <dfn>list of router rules</dfn> (a [=list=] of {{RouterRule}}s). It is initially an empty [=list=].
```

seems to match https://pr-preview.s3.amazonaws.com/sisidovski/ServiceWorkerForStaticRoutingAPI/pull/10.html#dfn-job-list-of-equivalent-jobs

domenic commented on docs/index.bs:
```diff
@@ -3193,7 +3202,8 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
                       1. Return |response|.
               1. Return null.
-      1. If |request| is a <a>non-subresource request</a>, |request| is a [=navigation request=], |registration|'s [=navigation preload enabled flag=] is set, |request|'s [=request/method=] is \`<code>GET</code>\`, |registration|'s [=active worker=]'s [=set of event types to handle=] [=set/contains=] <code>fetch</code>, and |registration|'s [=active worker=]'s [=all fetch listeners are empty flag ...
+          1. Assert: |source| is "fetch-event"
```
```suggestion
          1. Assert: |source| is "{{RouterSourceEnum/fetch-event}}"
```

domenic reviewed (commented):
Nothing substantial from me on this PR

yoshisatoyanagisawa committed:
Update docs/index.bs

Co-authored-by: Domenic Denicola <d@domenic.me>

yoshisatoyanagisawa committed:
Update docs/index.bs

Co-authored-by: Domenic Denicola <d@domenic.me>

yoshisatoyanagisawa committed:
Addressed comments.

- returns true.
- renamed to add "flag", but I still need to find a better name.

yoshisatoyanagisawa commented:
Although not all reviewers approved this PR, I am going to merge this so that people can see the full PR in https://github.com/w3c/ServiceWorker/pull/1701.
